libraries/AP_TemperatureSensor/AP_TemperatureSensor_MCP9600.h

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 2022  Kraus Hamdani Aerospace Inc. All rights reserved.
+ *
+ * This file is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This file is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Code by Tom Pittenger
+ */
+/*
+    Implements I2C driver for Microchip MCP9600 digital thermocouple EMF to Temperature converter
+*/
+
+#pragma once
+#include "AP_TemperatureSensor_Backend.h"
+
+#ifndef AP_TEMPERATURE_SENSOR_MCP9600_ENABLE
+    #define AP_TEMPERATURE_SENSOR_MCP9600_ENABLE AP_TEMPERATURE_SENSOR_ENABLED
+#endif
+
+#if AP_TEMPERATURE_SENSOR_MCP9600_ENABLE
+
+
+class AP_TemperatureSensor_MCP9600 : public AP_TemperatureSensor_Backend {
+    using AP_TemperatureSensor_Backend::AP_TemperatureSensor_Backend;
+public:
+
+    enum class ThermocoupleType : uint8_t {
```

Comment:
It could be nice to have a ThermoCoupleType  up higher as well.

Reply:
maybe.. lets discuss that in a later PR. Nothing higher up uses it right now. I think it might deserve a param too but lets deferr that for now